src/backend/distributed/utils/metadata_cache.c

```diff
@@ -339,10 +365,12 @@ CitusHasBeenLoaded(void)
 Oid
 DistShardRelationId(void)
 {
-	static Oid cachedOid = InvalidOid;
+	Oid cachedOid = metadataCacheState.distShardRelationId;
```

Comment:
Honestly it just looks nicer, I like the semantics of "this is all the state which needs to be cleared" which I can use to just `memset` the whole thing later.